examples/rosettacode/24_game.nit

```diff
@@ -0,0 +1,156 @@
+	var previous_char: nullable Char = null
+	var number: nullable Int = null
+	var negative = false
+
+	for i in operation.length.times do
+		var current_char = operation[i]
+		var current_int = current_char.to_i
+
+		if (previous_char == null or previous_char.is_op) and current_char == '-' then
+			negative = true
+			continue
+		end
+
+		if current_char.is_dig then
+			if number == null then
+				number = current_int
+			else
+				number = number * 10 + current_int
+			end
+		end
+
+		if negative and (current_char.is_op or i == operation.length - 1) then
+			number = number - number * 2
+			negative = false
+		end
+
+		if (current_char.is_op or i == operation.length - 1) and number != null then
+			numbers.add(number)
+			number = null
+		end
+
+		if not negative and current_char.is_op then
+			operands.add(current_char)
+		end
+		previous_char = current_char
+	end
+end
+
+# Create random numbers between 1 to 9
+fun random: Array[Int] do
+	return [for i in 4.times do 1 + 9.rand]
+end
+
+# Make mathematical operation with `numbers` and `operands` and add the operation result into `random_numbers`
+fun calculation(random_numbers, numbers: Array[Int], operands: Array[Char]) do
+	var number = 0
+	var temp_numbers = numbers.clone
+
+	while temp_numbers.length > 1 do
+		var operand = operands.shift
+		var a = temp_numbers.shift
+		var b = temp_numbers.shift
+
+		if operand == '+' then number = a + b
+		if operand == '-' then number = a - b
+		if operand == '*' then number = a * b
+		if operand == '/' then number = a / b
+
+		temp_numbers.unshift(number)
+	end
+	if number != 0 then random_numbers.add(number)
+end
+
+# Check if used `numbers` exist in the `random_numbers` created
+fun numbers_exists(random_numbers, numbers: Array[Int]): Bool do
+	for number in numbers do
+		if not random_numbers.count(number) >= numbers.count(number) then return false
+	end
+	return true
+end
+
+# Remove `numbers` when they are used
+fun remove_numbers(random_numbers, numbers: Array[Int]) do
+	for number in numbers do random_numbers.remove(number)
+end
+
+# Check if the mathematical `operation` is valid
+fun check(operation: String): Bool do
+	var previous_char: nullable Char = null
+	var next_char: nullable Char = null
+	var next_1_char: nullable Char = null
+
+	for i in operation.length.times do
+		var current_char = operation[i]
+
+		if i + 1 < operation.length then
+			next_char = operation[i + 1]
+			if i + 2 < operation.length then
+				next_1_char = operation[i + 2]
+			else
+				next_1_char = null
+			end
+		else
+			next_char = null
+		end
+
+		if not current_char.is_op and not current_char.is_dig then return false
+		if next_char == null and current_char.is_op then return false
+
+		if previous_char == null  then
+			if next_char == null or next_1_char == null then return false
+			if current_char == '-' and not next_char.is_dig then return false
+			if current_char != '-' and not current_char.is_dig then return false
+		end
+		if previous_char != null then
```

Comment:
You can merge this one with the next one.

~~~nit
if previous_char != null and next_char != null then ...
~~~